### DISCUSSION TITLE
Firefox Extensions

### DISCUSSION THREAD

**Discussion by BelugaByte** [seleniumbase/SeleniumBase#3258] [Q&A] [answered]
Hello,
first of all thanks for creating the project. 
Is it possible to install Firefox Extensions via SeleniumBase? With Chrome it was pretty easy.

**Comment by mdmintz** [comment, marked as answer]
Firefox calls them "Add-ons" (Chrome calls them "Extensions").
That's probably why you didn't find that in the docs (https://github.com/seleniumbase/SeleniumBase/issues/803#issuecomment-774402081).

See https://github.com/seleniumbase/SeleniumBase/releases/tag/v1.55.1
There's a method for loading ``.xpi`` files:

```python
self.install_addon(xpi_file)
```

For some reason, it must be done at runtime after the browser has already been spun up, unlike Chrome extensions, which must be initialized before the web browser is launched. So, load your Chrome extensions on the command-line, and load your Firefox add-ons inside the tests.
